### DISCUSSION TITLE
How do I export import image?

### DISCUSSION THREAD

**Discussion by drjoms** [containers/podman#21386] [Q&A] [answered]
I download some semi-random image from some red hat registry or docker one.
Run container. It works.
Then I save said image.
Then I import it.
Then I try to run it:
podman container run  ad9156f7bcd8
**Error: no command or entrypoint provided, and no CMD or ENTRYPOINT from image**

is there no simple way to make it work without 100+ obscure arguments?

I want to  learn how I can take image from registry, take it to offline machine WITHOUT registry access, and import it from tar file or something and ACTUALLY run it, with minimal arguments, etc.

podman info
host:
  arch: amd64
  buildahVersion: 1.31.3
  cgroupControllers:
  - memory
  - pids
  cgroupManager: systemd
  cgroupVersion: v2
  conmon:
    package: conmon-2.1.8-1.el9.x86_64
    path: /usr/bin/conmon
    version: 'conmon version 2.1.8, commit: cebaba63f66de0e92cdc7e2a59f39c9208281158'
  cpuUtilization:
    idlePercent: 99.83
    systemPercent: 0.09
    userPercent: 0.08
  cpus: 8
  databaseBackend: boltdb
  distribution:
    distribution: '"rocky"'
    version: "9.3"
  eventLogger: journald
  freeLocks: 2041
  hostname: VFIEVOX3.Router
  idMappings:
    gidmap:
    - container_id: 0
      host_id: 1000
      size: 1
    - container_id: 1
      host_id: 100000
      size: 65536
    uidmap:
    - container_id: 0
      host_id: 1000
      size: 1
    - container_id: 1
      host_id: 100000
      size: 65536
  kernel: 5.14.0-362.13.1.el9_3.x86_64
  linkmode: dynamic
  logDriver: journald
  memFree: 6603571200
  memTotal: 8045084672
  networkBackend: netavark
  networkBackendInfo:
    backend: netavark
    dns:
      package: aardvark-dns-1.7.0-1.el9.x86_64
      path: /usr/libexec/podman/aardvark-dns
      version: aardvark-dns 1.7.0
    package: netavark-1.7.0-2.el9_3.x86_64
    path: /usr/libexec/podman/netavark
    version: netavark 1.7.0
  ociRuntime:
    name: crun
    package: crun-1.8.7-1.el9.x86_64
    path: /usr/bin/crun
    version: |-
      crun version 1.8.7
      commit: 53a9996ce82d1ee818349bdcc64797a1fa0433c4
      rundir: /run/user/1000/crun
      spec: 1.0.0
      +SYSTEMD +SELINUX +APPARMOR +CAP +SECCOMP +EBPF +CRIU +YAJL
  os: linux
  pasta:
    executable: ""
    package: ""
    version: ""
  remoteSocket:
    path: /run/user/1000/podman/podman.sock
  security:
    apparmorEnabled: false
    capabilities: CAP_CHOWN,CAP_DAC_OVERRIDE,CAP_FOWNER,CAP_FSETID,CAP_KILL,CAP_NET_BIND_SERVICE,CAP_SETFCAP,CAP_SETGID,CAP_SETPCAP,CAP_SETUID,CAP_SYS_CHROOT
    rootless: true
    seccompEnabled: true
    seccompProfilePath: /usr/share/containers/seccomp.json
    selinuxEnabled: true
  serviceIsRemote: false
  slirp4netns:
    executable: /usr/bin/slirp4netns
    package: slirp4netns-1.2.1-1.el9.x86_64
    version: |-
      slirp4netns version 1.2.1
      commit: 09e31e92fa3d2a1d3ca261adaeb012c8d75a8194
      libslirp: 4.4.0
      SLIRP_CONFIG_VERSION_MAX: 3
      libseccomp: 2.5.2
  swapFree: 6333394944
  swapTotal: 6333394944
  uptime: 4h 39m 27.00s (Approximately 0.17 days)
plugins:
  authorization: null
  log:
  - k8s-file
  - none
  - passthrough
  - journald
  network:
  - bridge
  - macvlan
  - ipvlan
  volume:
  - local
registries:
  search:
  - registry.access.redhat.com
  - registry.redhat.io
  - docker.io
store:
  configFile: /home/dseletski/.config/containers/storage.conf
  containerStore:
    number: 5
    paused: 0
    running: 0
    stopped: 5
  graphDriverName: overlay
  graphOptions: {}
  graphRoot: /home/dseletski/.local/share/containers/storage
  graphRootAllocated: 32728154112
  graphRootUsed: 1639043072
  graphStatus:
    Backing Filesystem: xfs
    Native Overlay Diff: "true"
    Supports d_type: "true"
    Using metacopy: "false"
  imageCopyTmpDir: /var/tmp
  imageStore:
    number: 1
  runRoot: /run/user/1000/containers
  transientStore: false
  volumePath: /home/dseletski/.local/share/containers/storage/volumes
version:
  APIVersion: 4.6.1
  Built: 1702418000
  BuiltTime: Tue Dec 12 21:53:20 2023
  GitCommit: ""
  GoVersion: go1.20.10
  Os: linux
  OsArch: linux/amd64
  Version: 4.6.1

Thanks in advance.

**Comment by rhatdan** [comment]
If you inspect the old image and the new do the file look pretty much the same?

**Reply by drjoms** [reply]
Oh whoa, I did not expect to get a response from THE MAN himself!
I have read your book on Podman, clearly not well enough.

Please consider adding part where you can do save/load of an image :D
Because @nalind nailed it on a head!
Thanks for a cool book!

**Reply by drjoms** [reply]
is that Guinness in your hand? If you pass by Dublin, Ireland, ping me here - I will get you all Guinness you can drink ;)

**Reply by rhatdan** [reply]
That picture in on Shirken Island off of Baltimore in County Cork.  Visiting grandparents home.  But I prefer SmithWicks, but I am happy to have Guinness as well.

**Reply by drjoms** [reply]
Murphy's or O'Hara's or Beamish, offer still stands!
Take Nalin with you ;)

**Comment by nalind** [comment, marked as answer]
If you've used `podman save` to save the image, use `podman load` to load it elsewhere -- they use the same format, and the image's settings should be preserved.  Even if `podman import` doesn't error out when pointed at a file created by `podman save`, I don't think the result will be what you want.

**Comment by drjoms** [comment]
thank you Nalind, that was exactly what I was looking for.
Simple save/load. As opposed to save/import...
